Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/data/Array.java

```diff
@@ -111,6 +111,23 @@ default public String getArrayRepresentation(MemBuffer buf, Settings settings, i
 		return (value != null) ? value : "";
 	}
 
+	/**
+	 * Get the value object which corresponds to an array in memory.  This will either be a
+	 * String for the ArrayStringable case or null.
+	 * 
+	 * @param buf data buffer
+	 * @param settings data settings
+	 * @param length length of array
+	 * @return a String if it is an array of chars; otherwise null.
+	 */
+	public default StringDataInstance getStringDataInstance(MemBuffer buf, Settings settings, int length) {
+		if (!buf.getMemory().getAllInitializedAddressSet().contains(buf.getAddress())) {
```

Comment:
There *might* be a more appropriate way to check if the address is initialized memory other than collecting the entire set. You could probably peek at the implementation of `getAllInitializedAddressSet` for a hint here.

Reply:
Actually, this is copied directly from the `getArrayValue` down below. If we need a more appropirate way of do this, I think it maybe better to have another PR do this? Since actually.. I haven't checked and totally understood how this part works, but I think they share the same logic so I copied that here.

Reply:
I see. There is `Address.isLoadedMemoryAddress()` but I don't know if this means it is initialized or not. Any other alternative I can think of is more than one line.

Reply:
Are these 2 other classes necessary for this fix?

Reply:
Sorry this is the thing missing when I clean up code when the `data != null` block has been changed. This should be fine now.